core/src/avm2/globals/vector.rs

```diff
@@ -260,12 +260,15 @@ pub fn concat<'gc>(
             .subclass_object()
             .expect("Method call without bound class?");
         if !arg.is_of_type(activation, my_base_vector_class.inner_class_definition()) {
-            return Err(format!(
-                "TypeError: Cannot coerce argument of type {:?} to argument of type {:?}",
-                arg_class.read().name(),
-                my_base_vector_class.inner_class_definition().read().name()
-            )
-            .into());
+            return Err(Error::AvmError(type_error(
+                activation,
+                &format!(
+                    "Error #1034: Type Coercion failed: cannot convert {:?}@00000000000 to {:?}.",
```

Comment:
(long term we'll probably want to extract this into a function (not necessarily now), this error message is already duplicated...)
Oh BTW, the other implementation uses `instance_of_class_name()`, and `to_qualified_name_err_message()` are you sure that one isn't better?

Reply:
Yeah, those are more correct. I didn't extract it into a `make_error_1034` function- if it ends up being needed in more places, we should definitely do that.